src/audio/detect_test.c

```diff
@@ -337,23 +359,30 @@ static int test_keyword_set_model(struct comp_dev *dev,
 			return -EINVAL;
 		}
 
+		done = 1;
 		trace_keyword("test_keyword_set_model() "
 			      "final packet received");
 	}
 
-	if (cdata->data->size >
-	    cd->config.load_memory_size - cd->buf_copy_pos) {
+	if (cdata->num_elems >
+	    cd->model.data_size - cd->model.data_pos) {
 		trace_keyword_error("keyword_ctrl_set_model() error: "
 				   "too much data");
 		return -EINVAL;
 	}
 
-	assert(!memcpy_s(cd->load_memory + cd->buf_copy_pos,
-		 cd->config.load_memory_size - cd->buf_copy_pos,
-		 cdata->data->data, cdata->data->size));
+	assert(!memcpy_s(cd->model.data + cd->model.data_pos,
+			 cd->model.data_size - cd->model.data_pos,
+			 cdata->data->data, cdata->num_elems));
 
-	cd->buf_copy_pos += cdata->data->size;
+	cd->model.data_pos += cdata->num_elems;
 
+	if (done) {
+		/* Set model data done, update crc value */
+		cd->model.crc = crc32(cd->model.data, cd->model.data_size);
+		trace_keyword("done, memory_size:0x%x, crc: 0x%08x",
```

Comment:
please use the same trace message scheme as in other traces in this file

Reply:
sure, let me change it, only missed it after "memory_size:".

Reply:
@keyonjie not exactly, I just marked one, but there are a few more, line 335, 483, 492.

By "trace message scheme" I mean this string inside should use the same template like in other traces in this file and the whole project, also I forget about = instead of : 
so for example instead of:
tracev_keyword("msg_index:%d, num_elems:%d, remaining:%d "
it should be:
tracev_keyword("test_keyword_set_model(), msg_index = %d, num_elems = %d, remaining = %d "

small thing but we should use the same template to avoid mess, so just look at other traces and make it similar :)

Reply:
I actually don't like the scheme which specifying the whole function name, it is wasting dma trace bandwidth as we already has the source line number in each log item, and it will make you exceeding 80 characters very easy for the trace logs.

Reply:
But not big deal, let me change it to the unified format.